src/Microsoft.DotNet.Darc/src/DarcLib.AzDev/AzureDevOpsClient.cs

```diff
@@ -407,44 +407,49 @@ public async Task UpdatePullRequestAsync(string pullRequestUri, PullRequest pull
                 id);
         }
 
-        /// <summary>
-        ///     Merge a pull request
-        /// </summary>
-        /// <param name="pullRequestUrl">Uri of pull request to merge</param>
-        /// <param name="parameters">Settings for merge</param>
-        /// <returns></returns>
-        public async Task MergePullRequestAsync(string pullRequestUrl, MergePullRequestParameters parameters)
+        public async Task<IList<Commit>> GetPullRequestCommitsAsync(string pullRequestUrl)
         {
             (string accountName, string projectName, string repoName, int id) = ParsePullRequestUri(pullRequestUrl);
-
             VssConnection connection = CreateVssConnection(accountName);
             GitHttpClient client = await connection.GetClientAsync<GitHttpClient>();
-
-            string commitToMerge = parameters.CommitToMerge;
-
-            // If the commit to merge is empty, look it up first.
-            if (string.IsNullOrEmpty(commitToMerge))
+            var pullRequest = await client.GetPullRequestAsync(repoName, id, includeCommits: true);
+            IList<Commit> commits = new List<Commit>(pullRequest.Commits.Length);
+            foreach (var commit in pullRequest.Commits)
             {
-                var prInfo = await client.GetPullRequestAsync(repoName, id);
-                commitToMerge = prInfo.LastMergeSourceCommit.CommitId;
+                commits.Add(new Commit(commit.Author.Name == "DotNet-Bot" ? "dotnet-maestro[bot]" : commit.Author.Name,
+                    commit.CommitId,
+                    commit.Comment));
             }
+            connection.Dispose();
+            return commits;
+        }
+
+        public async Task MergeDependencyPullRequestAsync(string pullRequestUrl, MergePullRequestParameters parameters, string mergeCommit)
+        {
+            (string accountName, string projectName, string repoName, int id) = ParsePullRequestUri(pullRequestUrl);
+
+            VssConnection connection = CreateVssConnection(accountName);
+            GitHttpClient client = await connection.GetClientAsync<GitHttpClient>();
```

Comment:
These should be `using(GitHttpClient client = await connection.GetClientAsync<GitHttpClient>()) { ... }`